File: keiner_lindarte_aro/instructor_henry/mis_programas/practicas/dart/practicas_4/rifa.dart
import 'dart:io';
import 'dart:math';

void main () {

  int randnumber = Random().nextInt(10)+1;

  print('Cuantos participaran en la Rifa?');
  int cuantos = int.parse(stdin.readLineSync()!);

  for (int vueltas = 1; vueltas <= cuantos; vueltas++){

    int usernumber = Random().nextInt(10)+1;

    print('-----------------------------------------------------------------------------------------------------------------');

    print('Acontinuacion sacara un papel de una bolsa, donde puede haber numeros repetidos y dependiendo de lo que le salga, se gana o no la rifa.');

    print('-----------------------------------------------------------------------------------------------------------------');

    print('Pon "sacar" para sacar un papel de la bolsa:');
    String take = stdin.readLineSync()!;   

    print('-----------------------------------------------------------------------------------------------------------------');


    if (take == "sacar"){
      if (usernumber == randnumber){
        print('El usuario saco un papel con el numero $usernumber');
        print('Usted se ha ganado la rifa!!!');
        break;

      } else {
          print('El usuario saco un papel con el numero $usernumber');
          print('Mas suerte la proxima');
        } 
    } else {
      print('Tenias que sacar un papel del la bolsa, ahora pierdes tu participacion');
    }

    print('-----------------------------------------------------------------------------------------------------------------');

  }

  print('El numero ganador es $randnumber');

}

File: keiner_lindarte_aro/instructor_henry/mis_programas/practicas/ruby/practicas_4/rifa.rb
randnumber = rand(10)+1


puts "Cuantos participaran en la Rifa?"
cuantos = gets.chomp.to_i

for vueltas in 1..cuantos

    usernumber = rand(10)+1

    puts "-----------------------------------------------------------------------------------------------------------------"

    puts "Acontinuacion sacara un papel de una bolsa, donde puede haber numeros repetidos y dependiendo de lo que le salga, se gana o no la rifa."

    puts "-----------------------------------------------------------------------------------------------------------------"

    puts "Pon 'sacar' sacar un papel de la bolsa:"
    take = gets.chomp

    puts "-----------------------------------------------------------------------------------------------------------------"

    if take == "sacar"
        if usernumber == randnumber
            puts "El usuario saco un papel con el numero #{usernumber}"
            puts "Usted se ha ganado la rifa!!!"
            break
        else 
            puts "El usuario saco un papel con el numero #{usernumber}"
            puts "Mas suerte la proxima"
        end
    else
        puts "Tenias que sacar un papel del la bolsa, ahora pierdes tu participacion"
    end

    puts "-----------------------------------------------------------------------------------------------------------------"

end

puts "El numero ganador es #{randnumber}"